patches/api/0377-Update-Folder-Uses-Plugin-Name.patch

```diff
@@ -0,0 +1,78 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Xemorr <31805746+Xemorr@users.noreply.github.com>
+Date: Fri, 1 Apr 2022 19:57:40 +0100
+Subject: [PATCH] Update Folder Uses Plugin Name
+
+
+diff --git a/src/main/java/org/bukkit/plugin/SimplePluginManager.java b/src/main/java/org/bukkit/plugin/SimplePluginManager.java
+index 42da20011544075a9bea63a12ae86f2f21720667..45e694dfb43357f0beb37abdd0d889ec15f20e20 100644
+--- a/src/main/java/org/bukkit/plugin/SimplePluginManager.java
++++ b/src/main/java/org/bukkit/plugin/SimplePluginManager.java
+@@ -400,7 +400,7 @@ public final class SimplePluginManager implements PluginManager {
+     public synchronized Plugin loadPlugin(@NotNull File file) throws InvalidPluginException, UnknownDependencyException {
+         Validate.notNull(file, "File cannot be null");
+ 
+-        checkUpdate(file);
++        file = checkUpdate(file); // Paper - update the reference in case checkUpdate renamed it
+ 
+         Set<Pattern> filters = fileAssociations.keySet();
+         Plugin result = null;
+@@ -427,16 +427,53 @@ public final class SimplePluginManager implements PluginManager {
+         return result;
+     }
+ 
+-    private void checkUpdate(@NotNull File file) {
++    // Paper start - Update Folder Uses Plugin Name to replace
++    /**
++     * Replaces a plugin with a plugin of the same plugin name in the update folder.
++     * @param file
++     * @throws InvalidPluginException
++     */
++    private File checkUpdate(@NotNull File file) throws InvalidPluginException {
+         if (updateDirectory == null || !updateDirectory.isDirectory()) {
+-            return;
++            return file;
++        }
++        PluginLoader pluginLoader = getPluginLoader(file);
++        try {
++            String pluginName = pluginLoader.getPluginDescription(file).getName();
++            for (File updateFile : updateDirectory.listFiles()) {
++                if (updateFile.isFile()) {
```

Comment:
Early continues for this and the pluginname equals

Reply:
Hopefully I understood your review correctly :) changed made